libvast/vast/system/importer.hpp

```diff
@@ -56,15 +56,26 @@ struct importer_state {
 
   void send_report();
 
-  /// Stores the id offset for the next slice.
-  id next_id = 0u;
+  caf::error bump_boundary();
+
+  /// @returns the next unused id and increments the position by its argument.
+  id next_id(uint64_t advance);
```

Comment:
I feel you are mixing two things in this functions that should be separate, because the call site really confuses me.
1. `next_id()` or `current_id()` to return the next available ID.
2. `advance(n)` to move in the ID space.

Reply:
I started out with that separation, but realized that calling `next_id()` twice without `advance()` in between would be a logic error. For that reason I thought it would be better to force these 2 operations into lockstep by means of the API.

Reply:
That's a good point. Then the function just needs another name. How about calling it `assign(n)`? This makes it clear we are using `n` IDs an it's easy to communicate/see that the return value is the starting point for that.